ewatercycle/forcing/_pcrglobwb.py

```diff
@@ -66,13 +66,13 @@ def generate(  # type: ignore
                 "additional_datasets"
             ] = [DATASETS[dataset]]
 
-        basin = Path(shape).stem
+        basin = to_absolute_path(shape).stem
         recipe.data["diagnostics"]["diagnostic_daily"]["scripts"]["script"][
             "basin"
         ] = basin
 
         if extract_region is None:
-            extract_region = get_extents(shape)
+            extract_region = get_extents(to_absolute_path(shape))
```

Comment:
I think I would like it even better if we started the function with: `shape = to_absolute_path(shape)` so that we have all the `to_absolute_paths` close to the user inputs, rather than everywhere throughout the code.

Reply:
it is revised.